libs/community/langchain_community/utilities/astradb.py

```diff
@@ -46,39 +48,115 @@ def __init__(
+                token=token,
+                api_endpoint=api_endpoint,
+                namespace=self.namespace,
+            )
+            async_astra_db = AsyncAstraDB(
+                token=token,
+                api_endpoint=api_endpoint,
                 namespace=self.namespace,
             )
-            if supports_async:
-                self.async_astra_db = db.AsyncAstraDB(
-                    token=self.token,
-                    api_endpoint=self.api_endpoint,
-                    namespace=self.namespace,
-                )
 
         if astra_db:
             self.astra_db = astra_db
+            if async_astra_db:
+                self.async_astra_db = async_astra_db
+            else:
+                self.async_astra_db = AsyncAstraDB(
+                    token=self.astra_db.token,
+                    api_endpoint=self.astra_db.base_url,
+                    api_path=self.astra_db.api_path,
+                    api_version=self.astra_db.api_version,
+                    namespace=self.astra_db.namespace,
+                )
+        elif async_astra_db:
+            self.async_astra_db = async_astra_db
+            self.astra_db = AstraDB(
+                token=self.async_astra_db.token,
+                api_endpoint=self.async_astra_db.base_url,
+                api_path=self.async_astra_db.api_path,
+                api_version=self.async_astra_db.api_version,
+                namespace=self.async_astra_db.namespace,
+            )
+        else:
+            raise ValueError(
+                "Must provide 'astra_db_client' or 'async_astra_db_client' or "
+                "'token' and 'api_endpoint'"
+            )
+
+
+class AstraDBCollectionEnvironment(AstraDBEnvironment):
+    def __init__(
+        self,
+        collection_name: str,
+        token: Optional[str] = None,
+        api_endpoint: Optional[str] = None,
+        astra_db_client: Optional[AstraDB] = None,
+        async_astra_db_client: Optional[AsyncAstraDB] = None,
+        namespace: Optional[str] = None,
+        pre_delete_collection: bool = False,
+        async_setup: bool = False,
+        embedding_dimension: Union[int, Awaitable[int], None] = None,
+        metric: Optional[str] = None,
+    ) -> None:
+        from astrapy.db import AstraDBCollection, AsyncAstraDBCollection
+
+        super().__init__(
+            token, api_endpoint, astra_db_client, async_astra_db_client, namespace
+        )
+        self.collection_name = collection_name
+        self.collection = AstraDBCollection(
+            collection_name=collection_name,
+            astra_db=self.astra_db,
+        )
+
+        self.async_collection = AsyncAstraDBCollection(
+            collection_name=collection_name,
+            astra_db=self.async_astra_db,
+        )
+
+        self.async_setup_db_task: Optional[Task] = None
+        if async_setup:
+            async_astra_db = self.async_astra_db
+
+            async def _setup_db() -> None:
```

Comment:
The code block seems to force a network request during initialization.

I left a comment about this above. We don't necessarily need to do anything at this stage, but could be good to allow a developer to opt-out from the init somehow for more advanced use cases

Reply:
FYI, the Cassandra vector store is about to receive a `skip_provisioning: bool = False` flag for exactly that purpose. There might be a same-named parameter here.

Reply:
added SetupMode param (SYNC, ASYNC, OFF)